src/coreComponents/fileIO/vtk/VTKPolyDataWriterInterface.cpp

```diff
@@ -332,21 +331,127 @@ void VTKPolyDataWriterInterface::writeElementFields( vtkSmartPointer< vtkCellDat
     }
   } );
 
-  for( auto const & field : allFields )
+  for( string const & field : allFields )
   {
-    vtkSmartPointer< VTKGEOSXData > data = VTKGEOSXData::New();
-    data->SetNumberOfValues( er.getNumberOfElements< SUBREGION >() );
-    data->SetName( field.c_str() );
 
-    localIndex count = 0;
+    // Here is why we need a special treatment for some fields here.
+    // - Fields that have 3 dimensions or less can be treated in a purely generic fashion using the writeField function
+    // - Some fields have more than 3 dimensions (cell id, plus 3 others) and cannot be easily visualized in Paraview.
+    //   For now, we apply a special component-by-component treatment to 4-dimension fields (see below), but we do not output fields with 5
+    // dimensions or more.
+
+    // First, check if the field requires a special treatment. For now, array4d< real64 > (for instance, for phaseCompFraction) are
+    // treated separately to make sure that they can be easily visualized in Paraview.
+    bool is4DField = false;
     er.forElementSubRegions< SUBREGION >( [&]( auto const & esr )
     {
       WrapperBase const & wrapper = esr.getWrapperBase( field );
-      writeField( wrapper, data, esr.size(), count );
+      std::type_info const & typeID = wrapper.getTypeId();
+      if( typeID==typeid(array4d< real64 >) )
+      {
+        is4DField = true;
+      }
     } );
-    celldata->AddArray( data );
+
+    // Then, output the field to VTK
+    // Note: fields with 5 dimensions or more are not supported and are not output to file below
+    if( is4DField )
+    {
+      // If the field has four dimensions, we apply it a special treatment in the following function
+      writeElementField4D< SUBREGION >( celldata, er, field );
+    }
+    else
+    {
+      // If the field has three dimensions or less, it can be output in a generic fashion
+      writeElementField2D3D< SUBREGION >( celldata, er, field );
+    }
```

Comment:
So here we may be dealing with 1D?

Reply:
I renamed the function `writeElementField1D2D3D` to make it clear that this function is going to output the fields stored in `array1d`, `array2d`, and `array3d`.